File: DSRFQ.Web/Modules/ServerTypes/Wallet/WalletTransactionsForm.ts
import { IntegerEditor, LookupEditor, StringEditor, DecimalEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface WalletTransactionsForm {
    UserId: IntegerEditor;
    CompanyId: LookupEditor;
    StripeSessionId: StringEditor;
    StripePaymentIntentId: StringEditor;
    StripeInvoiceId: StringEditor;
    StripeInvoiceUrl: StringEditor;
    Amount: DecimalEditor;
    Currency: StringEditor;
    Status: StringEditor;
    PaymentStatus: StringEditor;
    CustomerName: StringEditor;
    CustomerEmail: StringEditor;
}

export class WalletTransactionsForm extends PrefixedContext {
    static readonly formKey = 'Wallet.WalletTransactions';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!WalletTransactionsForm.init)  {
            WalletTransactionsForm.init = true;

            var w0 = IntegerEditor;
            var w1 = LookupEditor;
            var w2 = StringEditor;
            var w3 = DecimalEditor;

            initFormType(WalletTransactionsForm, [
                'UserId', w0,
                'CompanyId', w1,
                'StripeSessionId', w2,
                'StripePaymentIntentId', w2,
                'StripeInvoiceId', w2,
                'StripeInvoiceUrl', w2,
                'Amount', w3,
                'Currency', w2,
                'Status', w2,
                'PaymentStatus', w2,
                'CustomerName', w2,
                'CustomerEmail', w2
            ]);
        }
    }
}

File: DSRFQ.Web/Modules/Wallet/WalletTransactions/WalletTransactionsForm.cs
using Serenity.ComponentModel;
using System;

namespace DSRFQ.Wallet.Forms;

[FormScript("Wallet.WalletTransactions")]
[BasedOnRow(typeof(WalletTransactionsRow), CheckNames = true)]
public class WalletTransactionsForm
{
    public int UserId { get; set; }
    public int CompanyId { get; set; }
    public string StripeSessionId { get; set; }
    public string StripePaymentIntentId { get; set; }
    public string StripeInvoiceId { get; set; }
    public string StripeInvoiceUrl { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; }
    public string Status { get; set; }
    public string PaymentStatus { get; set; }
    public string CustomerName { get; set; }
    public string CustomerEmail { get; set; }
  
}